Modules/FIT/FDD/src/DigitQcTask.cxx

```diff
@@ -823,6 +835,29 @@ void DigitQcTask::endOfCycle()
   // one has to set num. of entries manually because
   // default TH1Reductor gets only mean,stddev and entries (no integral)
   mHistCFDEff->Divide(mHistNumADC.get(), mHistNumCFD.get());
+  for (int iPM = 0; iPM < sNCHANNELS_PM; iPM++) {
+    double intNumerator = mHistAmp2Ch->ProjectionY("yNum", iPM + 1, iPM + 1)->Integral(mBinMinADCSaturationCheck, mBinMaxADCSaturationCheck);
+    double intDenominator = mHistAmp2Ch->ProjectionY("yDen", iPM + 1, iPM + 1)->Integral(mBinMinADCSaturationCheck, mHistAmp2Ch->GetNbinsY());
+    if (intDenominator)
+      mHistSaturationFraction->SetBinContent(iPM, intNumerator / intDenominator);
+  }
+
+  for (int channel = 0; channel <= sNCHANNELS_PM; channel++) {
+    float events_in_range = 0;
+    float events_per_channel = 0;
+    for (int bin_on_y_axis = 1; bin_on_y_axis <= mHistTime2Ch->GetNbinsY(); bin_on_y_axis++) {
+      if (mHistTime2Ch->GetYaxis()->GetBinLowEdge(bin_on_y_axis) > mMinTimeGate && mHistTime2Ch->GetYaxis()->GetBinLowEdge(bin_on_y_axis) < mMaxTimeGate) {
+        events_in_range += mHistTime2Ch->GetBinContent(channel + 1, bin_on_y_axis);
+      }
+      events_per_channel += mHistTime2Ch->GetBinContent(channel + 1, bin_on_y_axis);
+    }
+    if (events_per_channel) {
+      mHistGateTimeRatio2Ch->SetBinContent(channel + 1, events_in_range / events_per_channel);
+    } else {
+      mHistGateTimeRatio2Ch->SetBinContent(channel + 1, 0);
+    }
+  }
+  mHistSaturationFraction->GetYaxis()->SetRangeUser(0, 1.1);
```

Comment:
FIY this will not work if we use QC at EPN, better to move into PostProc. For FLP it is OK

Reply:
Hi at this moment, please keep it here, later we can move this to PostProcessing task

Reply:
I commented it just fyi, for keeping in mind if for some reason QC will be used at EPN farm

Reply:
Thank you very much